### PR TITLE
BigQuery: Add explicit service account key auth sample.

### PR DESCRIPTION
**Pull request by tswast**
For inclusion in the BigQuery authentication docs: https://cloud.google.com/bigquery/authentication

I followed the Python example here: https://cloud.google.com/docs/authentication/production#auth-cloud-explicit-python

### CODE REVIEW

File: bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ *
+ * <p>See: https://cloud.google.com/bigquery/authentication
+ */
+public class AuthSnippets {
+
+  // [START explicit_service_account]
+  public static void explicit() throws IOException {
+    // Load credentials from JSON key file.
+    GoogleCredentials credentials;
+    File credentialsPath = new File("service_account.json");  // TODO: update to your key path.
```
Comment (jabubake) [inline bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java]: Is setting the env var GOOGLE_APPLICATION_CREDENTIALS to point to the json file path not an option ?

Comment (lesv) [inline bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java]: I'd be happy if he just mentioned GOOGLE_APPLICATION_CREDENTIALS, I suspect the page mentions it.

File: bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    try (FileInputStream serviceAccountStream = new FileInputStream(credentialsPath)) {
+      credentials = ServiceAccountCredentials.fromStream(serviceAccountStream);
+    }
+
+    // Instantiate a client.
+    BigQuery bigquery =
+        BigQueryOptions.newBuilder().setCredentials(credentials).build().getService();
+
+    // Use the client.
+    for (Dataset dataset : bigquery.listDatasets().iterateAll()) {
```
Comment (lesv) [inline bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java]: Do you want to be symmetrical with the one below?  ("Datasets:")??

File: bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java (hunk 1 above)
Comment (tswast) [inline bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java]: It's an option, but this sample is for explicitly supplying it in the code. I've gotten feedback from Windows users who have trouble making the environment variable version work correctly.

I'll add a comment that mentions it.

File: bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java (hunk 2 above)
Comment (tswast) [inline bigquery/cloud-client/src/main/java/com/example/bigquery/AuthSnippets.java]: Added.

**Comment by tswast** [comment]
Added comments talking about `GOOGLE_APPLICATION_CREDENTIALS`.

**Comment by tswast** [comment]
@dpebot merge on green

**Pull request merged**